packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -980,11 +976,33 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return _getTextPositionVertical(offset, verticalOffset);
   }
 
-  /// Keeping [selection]'s [TextSelection.baseOffset] fixed, move the
-  /// [TextSelection.extentOffset] down by one line.
+  // Deletes the current uncollapsed [selection].
+  void _deleteSelection(SelectionChangedCause cause) {
```

Comment:
nit: A way to ensure the selection is not null at compile time is to take it as a method parameter instead of getting it from the instance variable. Also you'll probably be able to use less `!` this way.